hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java

```diff
@@ -87,11 +87,15 @@
   private static final Pattern URL_SCHEMA_PATTERN =
       Pattern.compile("([^\\.]+)\\.([^\\.]+)\\.{0,1}(.*)");
 
-  private static final String URI_EXCEPTION_TEXT = "Ozone file system URL " +
-      "should be one of the following formats: " +
-      "o3fs://bucket.volume/key  OR " +
-      "o3fs://bucket.volume.om-host.example.com/key  OR " +
-      "o3fs://bucket.volume.om-host.example.com:5678/key";
+  private String getUriExceptionText(Configuration conf) {
+    final String URI_EXCEPTION_TEXT = "Ozone file system URL " +
+        "should be one of the following formats: " +
+        "o3fs://bucket.volume/key  OR " +
+        "o3fs://bucket.volume.om-host.example.com/key  OR " +
+        "o3fs://bucket.volume.om-host.example.com:" +
+        OmUtils.getOmRpcPort(conf) + "/key";
```

Comment:
This might not work here, as here the configuration object can be not an instance of OzoneConfiguration, then ozone-site.xml and ozone-default.xml will not be added to resources.

The check is done inside BasicOzoneClientAdapterImpl.java 
https://github.com/apache/hadoop/blob/trunk/hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneClientAdapterImpl.java#L112

Reply:
will update the patch